### DISCUSSION TITLE
blog/what-statistical-test-should-i-do/

### DISCUSSION THREAD

**Discussion by giscus** [AntoineSoetewey/statsandr#93] [General]
# blog/what-statistical-test-should-i-do/

Select the most appropriate statistical (hypothesis) test based on the number of variables and their types with the help of a flowchart

https://statsandr.com/blog/what-statistical-test-should-i-do/

**Comment by MSHelm** [comment]
Awesome flowchart, thanks so much for preparing it! That's exactly what I was always looking for, kind of like GraphPad Prisms suggestions when doing statistics.

**Reply by AntoineSoetewey** [reply]
Thanks Martin for your feedback, glad you find it useful!

**Comment by BroVic** [comment]
This is a great job you've done, Antoine! Thank you very much!

**Reply by AntoineSoetewey** [reply]
Thanks Victor!

**Comment by PhilippPro** [comment]
Some years ago I made a similar overview, mainly for advicing medical researchers: 
http://philipppro.github.io/Statistical_tests_overview/
It is a table and not a tree structure. ;)

**Reply by AntoineSoetewey** [reply]
Thanks for your input Philipp.

Also nice to have it for plots! I keep it for the next time I need to explain which plot to draw.

**Comment by AntoineSoetewey** [comment]
For those who are interested in which plot is appropriate based on the type of their data, here are two visualizations:

1. https://www.data-to-viz.com/#explore
2. https://www.linkedin.com/posts/opensourcecommunity_data-datastorytelling-datascience-activity-6879744977417592832-9y4J

**Reply by PhilippPro** [reply]
Awesome website, especially the first!

**Reply by AntoineSoetewey** [reply]
Glad you like it Philipp!

**Comment by drfurtado** [comment]
Thanks for this post. It will be very helpful for my students.

**Reply by AntoineSoetewey** [reply]
Glad you find it useful Ovande!

**Comment by thomasvdb25** [comment]
Hello, I am preparing for a resit and I need some help, as i find it extremely difficult to find the right statistical test:
for example, this was one of my exam questions:
You are wondering how the purchasing channel, gender, and their interaction (CHANNEL*GENDER)
influence the evaluation of the shopping trip.
channel ( categorical) and gender (categorical) are the Independent variables, and the evaluation ( Continuous) is the dependent variable.
According to this flowchart, the test I should use should be one of the pink ones, right? As I have over 2 variables (channel, gender, evaluation)? However the answer key says I should use a two-way ANOVA, and I have no idea how to come up with that. Any help would be greatly appreciated.
(Mayybe it's important, but here's the full question:
a) State what your IVs and DV are.
b) Conduct the appropriate test. (No need to test for the model assumptions.)
c) Interpret the output (i.e., discuss the (non)significance of all of the predictors that you 
included)

**Reply by AntoineSoetewey** [reply]
Hello @thomasvdb25,

First, be careful that not *all* tests and methods are presented in the flowchart.

Since you have a continuous dependent variable and two categorical independent variables, you can choose between:

1. a multiple linear regression (presented in the flowchart),
1. or a two-way ANOVA (not presented in the flowchart).

Both methods require a continuous dependent variable. The difference between the two is that a two-way ANOVA requires that the two independent variables are categorical, whereas for the multiple linear regression the independent variables can be categorical and/or numerical (and are not limited to two independent variables).

Hope this helps.

Regards,
Antoine

**Comment by Abosamaha** [comment]
Antoine, your tremendous efforts are greatly appreciated.

**Reply by AntoineSoetewey** [reply]
Thank you @Abosamaha !

**Comment by Gilbert123-star** [comment]
I have 3 independent variables (2nominal and 1 ordinal) and 1 dependent variable (with just two possible outcome). what statistical techniques I will use?

**Reply by AntoineSoetewey** [reply]
Hello @Gilbert123-star,

If you follow the flowchart, you'll see that with a **dependent variable that is binary** (i.e., with only two outcomes), you can do a **binary logistic regression**.

I haven't written a tutorial about this method yet (I may write one in the future when I find the time), but you'll find plenty of resources online.

Regards,
Antoine

**Comment by Gilbert123-star** [comment]
Hi Antoine Soetewey, I really appreciate the flowchart you created on "statistics test should i do". It's such a helpful visual aid for understanding the concepts. I'm actually giving a presentation on a related topic in my RES1 class, and I think your flowchart would be a great addition to my slides. I wanted to ask for your permission to use the flowchart in my presentation. Of course, I would give you full credit as the creator of the flowchart, and I would only use it for educational purposes. If there are any specific conditions or requirements for using the flowchart, please let me know. Thank you.

**Reply by AntoineSoetewey** [reply]
Hello @Gilbert123-star,

Yes of course, feel free to use it!

Good luck for your presentation.

Regards,
Antoine

**Comment by musabohb** [comment]
extremely useful! exactly what i was looking for! thank you very much for taking the time to create this and to make it it publicly available! invaluable resource

**Reply by AntoineSoetewey** [reply]
Glad you find it useful @musabohb!

**Comment by rustacularApp** [comment]
Very well written article and flowchart. I always use this as my go-to for when i get any doubts on which test to use.

**Reply by AntoineSoetewey** [reply]
I’m glad you found it useful @h4ck4l1 !

**Reply by rustacularApp** [reply]
Sir if you would be kind enough to update the flow chart to include all the tests.i will be grateful..like ks test how and when to use...

**Reply by AntoineSoetewey** [reply]
By KS test I suppose you mean the Kolmogorov-Smirnov test?

If yes, note that this test is, in most cases, used to test normality (like the Shapiro-Wilk test).

Let me know if you have the same question for other tests.

**Comment by moi-taiga** [comment]
Super helpful resource, thank you Antoine!

**Reply by AntoineSoetewey** [reply]
Thank you @moi-taiga!
